composite-action/terragrunt/action.yml

```diff
@@ -155,7 +161,22 @@ runs:
         else
           echo "Changes detected. Redeploying everything..."
           terragrunt destroy -auto-approve --terragrunt-non-interactive
```

Comment:
I believe this is the issue that should be addressed. We should use `lifecycle.create_before_destroy` and not force-destroy the resource - it should get destroyed after the new one is created successfully, leaving it untouched in case of errors. What do you think? We can do it in a separate issue/PR.

Reply:
yeah makes sense. i will implement in this PR